### ISSUE TITLE
CrossGen2 nupkg on linux-musl-arm64 environment links to wrong libc

### ISSUE THREAD

**Issue by ayakael** [dotnet/runtime#66791]
### Description

CrossGen2 6.0.3 nupkg links to wrong libc on linux-musl-arm64 environment

### Reproduction Steps

```
$ wget 'https://globalcdn.nuget.org/packages/microsoft.netcore.app.crossgen2.linux-musl-arm64.6.0.3.nupkg'
$ unzip microsoft.netcore.app.crossgen2.linux-musl-arm64.6.0.3.nupkg
$ ldd tools/crossgen2
	/lib/ld-linux-aarch64.so.1 (0xffff8fce9000)
	libpthread.so.0 => /lib/ld-linux-aarch64.so.1 (0xffff8fce9000)
	libdl.so.2 => /lib/ld-linux-aarch64.so.1 (0xffff8fce9000)
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0xffff8fac8000)
	libm.so.6 => /lib/ld-linux-aarch64.so.1 (0xffff8fce9000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0xffff8faa3000)
	libc.so.6 => /lib/ld-linux-aarch64.so.1 (0xffff8fce9000)
Error loading shared library ld-linux-aarch64.so.1: No such file or directory (needed by /home/build/.nuget/packages/microsoft.netcore.app.crossgen2.linux-musl-arm64/6.0.3/tools/crossgen2)
```

### Expected behavior

Should link to `ld-linux-musl-aarch64.so.1`

### Actual behavior

Rather links to `ld-linux-aarch64.so.1`

### Known Workarounds

Does not occur when built via source-build on Alpine environment. See package available at via
`https://dl-cdn.alpinelinux.org/alpine/edge/testing/armv7/dotnet6-bootstrap-artifacts-6.0.102-r1.apk` (tar package, availabe at `./usr/share/dotnet/artifacts/6.0.102/microsoft.netcore.app.crossgen2.linux-musl-arm64.6.0.2.nupkg`)

**Comment by mangod9** [comment]
Thanks for reporting the issue @ayakael. Is this a new issue on 6.0.3?

**Comment by mangod9** [comment]
@janvorli has anything changed recently on musl builds?

**Comment by janvorli** [comment]
No, nothing has changed. This looks like some higher level RID related issue.

**Comment by agocke** [comment]
I don't think this is a RID issue -- the actual binary inside the package is linked to the wrong libc. I think either our packaging is wrong (it's pulling from the wrong directory) or the build is wrong (it's building against the wrong libc).

**Comment by am11** [comment]
It is the case of `CrossBuild = true` where we were computing wrong RID for crossgen, and eventually picking up linux-arm64 apphost instead of linux-musl-arm64. Fixed by #66814.

**Comment by am11** [comment]
@ayakael, this issue should be fixed in latest `main` now that the bigger refactoring PR https://github.com/dotnet/runtime/pull/65948  is merged. Please let me know if that's not the case. Thanks.
